WriteBack/ListTriggerAsync.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using System.Collections;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Redis;
+using System.Collections.Generic;
+using StackExchange.Redis;
+using System.Linq;
+using Microsoft.Azure.Cosmos.Linq;
+using Microsoft.Extensions.Configuration;
+using System.IO;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Redis.Samples
+{
+    public record ListData
+    (
+        string id,
+        List<string> value
+    );
+    public static class WriteBack
+    {
+        //Redis Cache primary connection string from local.settings.json
+        public const string localhostSetting = "redisLocalhost";
+        private static readonly IDatabase cache = ConnectionMultiplexer.Connect(Environment.GetEnvironmentVariable(localhostSetting)).GetDatabase();
+
+        //CosmosDB Endpoint from local.settings.json
+        public const string Endpoint = "Endpoint";
+
+        //CosmosDB database name and container name declared here
+        public const string databaseName = "databaseName";
+        public const string containerName = "containerName";
+
+        //Uses the key of the user's choice and should be changed accordingly
+        public const string key = "listTest";
+
+        [FunctionName(nameof(ListTriggerAsync))]
+        public static async Task ListTriggerAsync(
+            [RedisListTrigger(localhostSetting, key)] string listEntry, [CosmosDB(
+            Connection = "Endpoint" )]CosmosClient client,
+            ILogger logger)
+        {
+            //Retrieve the database and container from the given client, which accesses the CosmosDB Endpoint
+            Container db = client.GetDatabase(databaseName).GetContainer(containerName);
+
+            //Creates query for item inthe container and
+            //uses feed iterator to keep track of token when receiving results from query
+            IOrderedQueryable<ListData> query = db.GetItemLinqQueryable<ListData>();
```

Comment:
Would this work as well? 
```
IOrderedQueryable<ListData> orderedQuery = db.GetItemLinqQueryable<ListData>();
ListData item = orderedQuery
    .Where(p => p.id == key)
    .FirstOrDefault(defaultValue: null);
```

Reply:
Can you explain why the feed iterator is needed or what types of errors you were seeing? I think we should understand exactly why we need it if we add it here.

Reply:
the feed iterator is apart of cosmosdb sdk and it lets us get a list of objects from the database., so that's why it was my first option to use. When I changed to what you suggested and then removed lines 49 and 50, my error was suggesting to use GetItemQueryIterator in order to execute asynchronously.

Reply:
I see, thanks